src/main/java/org/folio/rest/impl/InvoicesImpl.java

```diff
@@ -167,7 +170,57 @@ public void getInvoiceInvoiceNumber(String lang, Map<String, String> okapiHeader
     asyncResultHandler.handle(succeededFuture(GetInvoiceInvoiceNumberResponse.respond500WithTextPlain(NOT_SUPPORTED)));
   }
 
+  @Validate
+  @Override
+  public void postInvoiceInvoicesDocumentsById(String id, String lang, InvoiceDocument entity, Map<String, String> okapiHeaders,
+    Handler<AsyncResult<Response>> asyncResultHandler, Context vertxContext) {
+    DocumentHelper documentHelper = new DocumentHelper(okapiHeaders, vertxContext, lang);
+    if (!entity.getDocumentMetadata().getInvoiceId()
+      .equals(id)) {
+      documentHelper.addProcessingError(MISMATCH_BETWEEN_ID_IN_PATH_AND_BODY.toError());
+      asyncResultHandler.handle(succeededFuture(documentHelper.buildErrorResponse(422)));
+    } else {
+      documentHelper.createDocument(id, entity)
+        .thenAccept(document -> {
+          logInfo("Successfully created document with id={}", id);
+          asyncResultHandler.handle(succeededFuture(documentHelper.buildResponseWithLocation(String.format(DOCUMENTS_LOCATION_PREFIX, id, document.getDocumentMetadata().getId()), document)));
+        })
+        .exceptionally(t -> handleErrorResponse(asyncResultHandler, documentHelper, t));
+    }
+  }
+
+  @Validate
+  @Override
+  public void getInvoiceInvoicesDocumentsById(String id, int offset, int limit, String query, String lang, Map<String, String> okapiHeaders, Handler<AsyncResult<Response>> asyncResultHandler, Context vertxContext) {
+    DocumentHelper documentHelper = new DocumentHelper(okapiHeaders, vertxContext, lang);
+    documentHelper.getDocumentsByInvoiceId(id)
```

Comment:
The `query` should be taken into account (as well as `offset` and `limit`).

Reply:
implementation fixed